Signal/src/ViewControllers/ContactViewController.swift

```diff
@@ -610,6 +549,26 @@ class ContactViewController: OWSViewController, CNContactViewControllerDelegate
         inviteFlow.sendSMSTo(phoneNumbers: phoneNumbers)
     }
 
+    func didPressAddToContacts() {
+        Logger.info("\(logTag) \(#function)")
+
+        let actionSheet = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+
+        actionSheet.addAction(UIAlertAction(title: NSLocalizedString("CONVERSATION_SETTINGS_NEW_CONTACT",
+                                                                     comment: "Label for 'new contact' button in conversation settings view."),
+                                            style: .default) { _ in
+                                                self.didPressCreateNewContact()
+        })
+        actionSheet.addAction(UIAlertAction(title: NSLocalizedString("CONVERSATION_SETTINGS_ADD_TO_EXISTING_CONTACT",
+                                                                     comment: "Label for 'new contact' button in conversation settings view."),
+                                            style: .default) { _ in
+                                                self.didPressAddToExistingContact()
+        })
```

Comment:
Myles decided to make this multi-step so that it's a consistent experience here and in conversation view.